### PR TITLE
4/get schema by version

### LINKED ISSUE #4: Add utils to get appData schema json by version
Proposed by Nenad https://github.com/cowprotocol/cow-sdk/pull/50#discussion_r924551112

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Closes #4 

Builds on top of #5 

Added `getAppDataSchema` utils method

# Testing

Unit tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "compile": "ts-node --esm src/scripts/compile.ts --experimental-specifier-resolution=node",
     "build": "yarn clean && yarn compile && microbundle",
     "test": "jest",
-    "prepare": "npm run build"
+    "prepare": "yarn build && yarn test"
```
Comment (ramirotw) [inline package.json:24]: nice, I believe we have this same problem in the sdk, mixing yarn and npm. It has issues when using stuff like nvm and you can have different nodejs versions when npm is used through a script as it uses a cached instance which is different than the current instance

**Review by ramirotw — APPROVED** [review]

File: src/utils.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+import {
+  AnyAppDataDocVersion,
+  latest,
+  LATEST_APP_DATA_VERSION,
+  LATEST_QUOTE_METADATA_VERSION,
+  LATEST_REFERRER_METADATA_VERSION,
+} from './generatedTypes'
 
 // TODO: Cannot understand why this doesn't work. TS complains saying the type parameters are unknown
 // export function createQuoteMetadata(params: Omit<latest.Quote, 'version'>): latest.Quote {
```
Comment (W3stside) [inline src/utils.ts:10]: try `latest['Quote'] ? not sure what else to suggest without looking at it. Isn't TS ignoring `generatedTypes` also? probably why

File: src/utils.ts [hunk 3]
```diff
@@ -27,6 +33,17 @@ export function createAppDataDoc(params: {
...
-}
\ No newline at end of file
+}
+
+export async function getAppDataSchema(version: string): Promise<AnyAppDataDocVersion> {
+  if (!/^\d+\.\d+\.\d+$/.test(version)) {
+    throw new Error(`AppData version ${version} is not a valid version`)
+  }
+  try {
+    return await import(`../schemas/v${version}.json`)
```
Comment (W3stside) [inline src/utils.ts:45]: just return it

**Review by W3stside — APPROVED** [review]

File: src/utils.ts (hunk 3 above)
Comment (alfetopito) [reply src/utils.ts:45]: If I do that unit tests fail.
Don't want to spend time to investigate why

Screen Shot 2022-07-22 at 10 50 12

File: src/utils.ts (hunk 2 above)
Comment (alfetopito) [reply src/utils.ts:10]: Screen Shot 2022-07-22 at 10 52 11

Screen Shot 2022-07-22 at 10 53 00

**Pull request merged**